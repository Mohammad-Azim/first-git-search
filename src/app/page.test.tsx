import { render } from "@testing-library/react";
import Home from "./page";

jest.mock("../components/search/Search", () => () => <div>Search Component</div>);
jest.mock("../components/search/SearchResultComponent", () => () => (
	<div>SearchResult Component</div>
));

describe("Home", () => {
	it("renders all the required components", () => {
		const { container } = render(<Home />);

		expect(container.textContent).toContain("Search Component");
		expect(container.textContent).toContain("SearchResult Component");
	});

	it("matches the snapshot", () => {
		const { container } = render(<Home />);

		expect(container).toMatchSnapshot();
	});
});
