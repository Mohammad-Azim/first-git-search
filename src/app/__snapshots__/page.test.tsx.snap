// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-2730jv-MuiGrid-root"
  >
    <div>
      Search Component
    </div>
    <div>
      SearchResult Component
    </div>
  </div>
</div>
`;
